app-creator/client/src/widgets/app-root.ts

```diff
@@ -93,19 +95,64 @@ export class AppRoot extends LitElement {
       font-weight: 500;
       color: var(--accent-color);
     }
+
+    paper-progress {
+      width: 100%;
+      height: 20px;
+      z-index: 10;
+      position: absolute;
+      top: 0;
+      left: 0;
+      margin: 0px;
+    }
   `;
 
   /**
    * Sets template dialog search query.
    */
   @property({ type: String }) query = '';
 
+  /**
+   * Array of templates.
+   */
+  @property({ type: Array }) templates: TemplateItem[] = [];
+
+  /**
+   * Loading state.
+   */
+  @property({ type: Boolean }) loading = false;
+
   /**
    * Reference to dialog element.
    */
   @query('paper-dialog') dialog!: PaperDialogElement;
 
+  async fetchTemplates() {
```

Comment:
Better to move the fetch logic up to the `TemplateManager` and implement `getTemplate` as async (performing the the fetch if `templates` is empty), so that consumers of `getTemplate` will always block until the fetch is complete (or the cached results are returned). I would also support a parameter `forceReload` to force the fetch to happen again.